x-pack/plugins/grokdebugger/public/plugin.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { first } from 'rxjs/operators';
+import { registerFeature } from './register_feature';
+import { PLUGIN } from '../common/constants';
+
+const inactiveLicenseMessage = i18n.translate('xpack.grokDebugger.clientInactiveLicenseError', {
+  defaultMessage: 'The Grok Debugger tool requires an active license.',
+});
+
+export class Plugin {
+  setup(coreSetup, plugins) {
+    registerFeature(plugins.home);
+
+    this.licensing = plugins.licensing;
+
+    this.licensing.license$.pipe(first()).subscribe(license => {
+      plugins.devTools.register({
+        order: 6,
+        title: i18n.translate('xpack.grokDebugger.displayName', {
+          defaultMessage: 'Grok Debugger',
+        }),
+        id: PLUGIN.ID,
+        enableRouting: false,
+        disabled: !license.isActive,
+        tooltipContent: !license.isActive ? inactiveLicenseMessage : null,
+        async mount(context, { element }) {
+          const [coreStart] = await coreSetup.getStartServices();
```

Comment:
I noticed that the `/kibana#/dev_tools/grokdebugger` URL still works even if the license isn't active. How about we return early here in this case, to prevent the app from rendering? It's not an elegant solution, but I think it's OK for now.

```js
        async mount(context, { element }) {
          if (!license.isActive) {
            // Users can still visit the app by hitting the route directly, so we'll just stop
            // the app from rendering for now.
            return;
          }
          const [coreStart] = await coreSetup.getStartServices();
          const { renderApp } = await import('./render_app');
          return renderApp(element, coreStart);
        },
```

Reply:
Yep, good point 👍 The [old logic](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/grokdebugger/public/register.js#L26) showed a toast and redirected. I'll reimplement that logic. Again will also link in with @jloleysens feedback about plugin registration.